### ISSUE TITLE
`stats.warningsFilter` support

### ISSUE THREAD

**Issue by mf-pherlihy** [taehwanno/warnings-to-errors-webpack-plugin#60]
Only Webpack 5 supports `{ ignoreWarnings }` as a configuration option.

Webpack 4 uses https://v4.webpack.js.org/configuration/stats/#statswarningsfilter which doesnt seem to be supported by this plugin.

I have tried using `new LoaderOptionsPlugin({ options: { ignoreWarnings: [...] } })` and `stats.warningsFilter = [...]` without much luck. 

Is there anyway we can support both cases?

**Comment by taehwanno** [comment]
As can be seen from #51 comment, I think it should improve internal implementation. but it has been delayed due to my personal priorities. but I plan to work on it. maybe this weekend?

**Comment by taehwanno** [comment]
It was published as v2.3.0 :)

**Comment by mf-pherlihy** [comment]
So this works if were targeting the warning message, but NOT the module id or context that triggered the warning.

Example Warning Object:

```
{
  message: 'Critical dependency: the request of a dependency is an expression',
  module: {
    context: '/local-path/node_modules/yargs',
	id: './node_modules/yargs/index.cjs',
  },
}
```

```javascript
// Does NOT work
webpackOptions.stats.warningsFilter: [/node_modules\/yargs/i];

// Works
webpackOptions.stats.warningsFilter: ['the request of a dependency is an expression'];
```

Note: If this plugin is not used, the `[/node_modules\/yargs/i]` warning filter works fine -- they are just not transitioned to errors.
